oclif/plugin-plugins
# Sm/linked-plugin-compile

during plugin:link, if it looks like a TS plugin, we'll add a compilation step.

This looks at devDeps to detect TS.  not only is that a decent proxy for "is it written in TS" but it also lets use safely compile via tsc because the devDep includes that.

fix https://github.com/forcedotcom/cli/issues/1664

> this does not fix the issues of ts plugins not recompiling on each run, or oclif using the compiled code for linked plugins instead of the source

QA ideas:
clone this, switch this branch, then you can link plugin-plugins into your cli `sfdx plugins:link .`

so that you get something like
```
sfdx plugins
@oclif/plugin-plugins 2.1.7 (link) /Users/shane.mclaughlin/eng/oclif-repos/plugin-plugins
```

Then you can link any other plugin you have locally.  If you delete the `lib` folder, when you run the link command, it should be recreated for you. 

@W-11731121@
